keps/NNNN-kep-template/kep.yaml

```diff
@@ -0,0 +1,23 @@
+title: KEP Template
+authors:
+  - "@janedoe"
+owning-sig: sig-xyz
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
```

Comment:
I think status is the status of the KEP, not the enhancement?  At least , that's how I read it.  Now, maybe we should consider this overtly.  Specifically, can a KEP become "implementable" with nobody attached to it?  What happens if I submit a KEP with honest intentions to implement it, and I lose interest, find a new job, come down with amnesia, or get hit by a bus?  Is the KEP still implementable?  Should it be deleted if nobody is actively working on it?  We have nothing here that talks about ownership of the work to implement a KEP.  Just the author of the KEP and the reviewers.

I thought the progress of the implementation was to be covered by the enhancement issue (or something else), not this KEP.  am I wrong?

I think deferred is only useful if KEPs track implementation.

Given the "merge and iterate" ethos, would we `git rm` KEPs that we decide not to do?  Or should we clarify the lifecycle such that you DO NOT merge until you remove all risk of being rejected?  Assuming you got past that hurdle, and then some new info arose, should teh KEP stay and be "withdrawn" or should it be `git rm` ?

I assume that "replaced" means "not fully implemented, but superseded".  Without this status, what happens to the original KEP?  Does it languish as "implementable" or "provisional"?  Or do we `git rm` ?

I want less states, but to do that we need to understand the lifecycle.

Reply:
We should consider adding alpha/beta/GA with the releases they are targeted for. This would allow tooling to check the milestone. Also FeatureGate. But those can come in a follow up PR, maybe after the discussion at the summit.

Reply:
> Add in `alpha|beta|ga`.

I'm in favor of the long term move to keep track of KEP state (`alpha`/`beta`/`ga`) in the metadata with the goal of moving more of that information into something machine consumable. 

> Specifically, can a KEP become "implementable" with nobody attached to it?

I think so. In general, my interpretation has been a KEP is implementable when all parties have agreed on the design and plan. It may not be done right now, but the plan is sound. I know this is intended to be `deferred` but I haven't seen folk using it.

This isn't perfect, but to get a little bit of a pulse on previous status usage:
```
git --no-pager log --online -S 'status: <status>' keps/ | wc -l
```
- provisional: 156
- implementable: 170
- implemented: 29
- deferred: 0
- rejected: 0
- withdrawn: 0
- replaced: 0

I think most of them have a place...the one that I think could be pulled would be `deferred`. Usually things fall by the wayside over time, folks will rarely go back to update the status. =/ An alternative might be to have the enhancements team step in to set it on ones that have routinely slipped or not acked to being included.

Reply:
Let me open an issue for this. It's a larger discussion.
I think it makes sense to bind the KEP's status to the enhancement's implementation status. Again, this is a thing that would make it easier for other systems consuming the information to do so w/o inference.

Reply:
For this PR, I'm going to punt on this, then.